File: uniseek_java/src/main/java/com/uniseek/entity/Enterprise.java
package com.uniseek.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 企业资质认证实体
 */
@Data
@TableName("enterprise")
public class Enterprise {

    /** 企业记录 ID */
    @TableId
    private Long id;

    /** 用户 ID（关联 user 表） */
    private Long userId;

    /** 企业名称 */
    private String companyName;

    /** 统一社会信用代码 */
    private String creditCode;

    /** 营业执照图片 URL */
    private String licenseImgUrl;

    /** 所属行业 */
    private String industry;

    /** 企业所在地区ID */
    @TableField("region_id")
    private Long regionId;

    /** 企业描述 */
    private String description;

    /** 审核状态：0 待审核 / 1 审核通过 / 2 审核不通过 */
    private Integer auditStatus;

    /** 企业认证通过时间 */
    @TableField("audit_time")
    private LocalDateTime auditTime;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: uniseek_vue/src/api/enterprise.ts
import request from './index'
import type { ApiResponse } from './auth'
import type { PageResult } from './task'

// 企业资质认证请求参数（匹配后端 EnterpriseRequest）
export interface EnterpriseRequest {
  companyName: string
  creditCode: string
  licenseImgUrl: string
  industry: string
  regionId: number | null
  description: string
}

// 企业信息（匹配后端 Enterprise 实体）
export interface EnterpriseInfo {
  id: number
  userId: number
  companyName: string
  creditCode: string
  licenseImgUrl: string
  industry: string
  regionId: number | null
  description: string
  auditStatus: number
  auditTime: string | null
  createTime: string
  updateTime: string
}

/** POST /api/enterprise 提交企业资质认证 */
export const submitEnterprise = async (params: EnterpriseRequest) => {
  const res = await request.post<ApiResponse<EnterpriseInfo>>('/enterprise', params)
  return res
}

/** GET /api/enterprise/my 查询我的企业资质信息 */
export const getMyEnterprise = async () => {
  const res = await request.get<ApiResponse<EnterpriseInfo>>('/enterprise/my')
  return res
}

/** PUT /api/enterprise 更新企业资质 */
export const updateEnterprise = async (params: EnterpriseRequest) => {
  const res = await request.put<ApiResponse<EnterpriseInfo>>('/enterprise', params)
  return res
}

/** GET /api/enterprise/list 分页获取已认证的企业列表 */
export interface EnterpriseListParams {
  page?: number
  pageSize?: number
  keyword?: string
  industry?: string
  regionId?: number
  sortBy?: string
  sortOrder?: string
}
export const getEnterpriseList = (params: EnterpriseListParams = {}) =>
  request.get<any, PageResult<EnterpriseInfo>>('/enterprise/list', { params })

// 热门企业信息
export interface HotEnterprise {
  id: number
  companyName: string
  industry: string
  regionId: number | null
  regionName: string
  description: string
  heatScore: number
  activeJobCount: number
}

/** GET /api/enterprise/hot 获取热门企业列表 */
export const getHotEnterprises = (limit = 12) =>
  request.get<any, HotEnterprise[]>('/enterprise/hot', { params: { limit } })

/** GET /api/enterprise/{id} 根据ID获取企业详情 */
export const getEnterpriseById = (id: number) =>
  request.get<any, EnterpriseInfo>(`/enterprise/${id}`)
